tests/distances/test_scphylo_wrapper.py

```diff
@@ -63,3 +70,164 @@ def test_another_is_different_and_symmetric(similarity, model_tree) -> None:
         assert similarity.calculate(model_tree, root) == pytest.approx(
             similarity.calculate(root, model_tree)
         )
+
+
+def tree_gen(tree_type: str, seed: int) -> Callable[[int], TreeNode]:
+    """Return a tree."""
+    rng = jax.random.PRNGKey(seed)
+
+    def tree1_(n_nodes: int) -> TreeNode:
+        """Return a tree."""
+        if tree_type == "r":
+            return ti.generate_random_TreeNode(rng, n_nodes)
+        elif tree_type == "d":
+            return ti.generate_deep_TreeNode(rng, n_nodes)
+        else:  # tree_type == "s"
+            return ti.generate_star_TreeNode(n_nodes)
+
+    return tree1_
+
+
+@pytest.mark.parametrize("tree_type1", ["r", "d", "s"])
+@pytest.mark.parametrize("seed1", [76, 42])
+@pytest.mark.parametrize("tree_type2", ["r", "d", "s"])
+@pytest.mark.parametrize("seed2", [13, 42])
+@pytest.mark.parametrize("n_nodes", [5, 10])
+def test_MP3Similarity(n_nodes: int, tree_type1, seed1: int, tree_type2, seed2: int):
+    """Test the MP3Similarity class."""
+
+    tree1_fn = tree_gen(tree_type=tree_type1, seed=seed1)
+    tree2_fn = tree_gen(tree_type=tree_type2, seed=seed2)
+
+    tree1 = tree1_fn(n_nodes)
+    tree2 = tree2_fn(n_nodes)
+
+    sim = dist.MP3Similarity()
+
+    try:
+        result = sim.calculate(
+            TreeNode.convert_to_anytree_node(tree1),
+            TreeNode.convert_to_anytree_node(tree2),
+        )
+        print("\n")
+        tree1.print_topo()
+        tree2.print_topo()
+        print(f"MP3: {result}")
+    except Exception as e:
+        print(e)
+        tree1.print_topo()
+        tree2.print_topo()
+        raise e
+
+
+# TODO: (Gordon) Reactivate test if scyphylo metric is fixed.
+@pytest.mark.skip("scyphylo metric under investigation")
+@pytest.mark.parametrize("tree_type1", ["r", "d", "s"])
+@pytest.mark.parametrize("seed1", [76, 42])
+@pytest.mark.parametrize("tree_type2", ["r", "d", "s"])
+@pytest.mark.parametrize("seed2", [13, 42])
+@pytest.mark.parametrize("n_nodes", [5, 10])
+def test_AncestorDescendantSimilarity(
+    n_nodes: int, tree_type1, seed1: int, tree_type2, seed2: int
+):
+    """Test the AncestorDescendantSimilarity scyphylo class."""
```

Comment:
Failing test - `scyphylo` just throws `DivisionByZero` errors, every about 5 trees.

Reply:
(Let's leave scphylo's AD, then. Apologies for this issue, I should've checked their test coverage before proposing this package!)